service/frontend/workflow_handler.go

```diff
@@ -4979,3 +5022,46 @@ func (wh *WorkflowHandler) unaliasCreateScheduleRequestSearchAttributes(request
 
 	return &newRequest, nil
 }
+
+func (wh *WorkflowHandler) getScheduleExecutionInfoAndSchedulingArgs(ctx context.Context, workflowID string, namespace string, namespaceID namespace.ID) (*workflowpb.WorkflowExecutionInfo, *schedspb.StartScheduleArgs, error) {
+	execution := &commonpb.WorkflowExecution{WorkflowId: workflowID}
+	// describe to get memo, search attributes, and execution info
+	describeResponse, err := wh.historyClient.DescribeWorkflowExecution(ctx, &historyservice.DescribeWorkflowExecutionRequest{
+		NamespaceId: namespaceID.String(),
+		Request: &workflowservice.DescribeWorkflowExecutionRequest{
+			Namespace: namespace,
+			Execution: execution,
+		},
+	})
+	if err != nil {
+		// TODO: rewrite "workflow" in error messages to "schedule"
+		return nil, nil, err
+	}
+
+	executionInfo := describeResponse.GetWorkflowExecutionInfo()
+	var startScheduleArgs schedspb.StartScheduleArgs
+	switch executionInfo.GetStatus() {
+	case enumspb.WORKFLOW_EXECUTION_STATUS_COMPLETED:
+		reverseHistory, err := wh.GetWorkflowExecutionHistoryReverse(ctx, &workflowservice.GetWorkflowExecutionHistoryReverseRequest{
+			Namespace:       namespace,
+			Execution:       execution,
```

Comment:
should probably add the run id we just found here.. I think it's more efficient that way